pyanaconda/network.py

```diff
@@ -688,6 +702,15 @@ def add_connection_for_ksdata(networkdata, devname):
     added_connections.insert(0, (con_uuid, dev_spec))
     return added_connections
 
+def connection_binding_setting(devname, bindto):
+    if bindto == BIND_TO_MAC:
+        hwaddr = nm.nm_device_perm_hwaddress(devname)
+        hwaddr = [int(b, 16) for b in hwaddr.split(":")]
+        setting = ['802-3-ethernet', 'mac-address', hwaddr, 'ay']
+    else:
+        setting = ['connection', 'interface-name', devname, 's']
+    return setting
```

Comment:
Wouldn't it be better to have two methods here:

`connection_binding_setting()` and `connection_binding_to_mac_setting()` instead of the flag passed in?

Reply:
Because it is the same right now. This is just a composition of two functions. The only line which will be called in both cases is the condition and this condition can be also outside or somewhere else. And in general it is simpler to test separated functions than composition of functions.

So when the code is not shared it is more descriptive to have these as two methods which information what is changed.

Of course this is more recommendation than requirement.

Reply:
Then they should be named
connection_binding_to_interface_name_setting() and connection_binding_to_mac_setting()
or better
bind_to_interface_name_connection_setting() and bind_to_mac_connection_setting()
or
bind_to_interface_name_setting() and bind_to_mac_setting()

Having two such functions doesn't make sense to me.

If I wanted to move the bindto argument branching one level up out of the function, I'd just create a function to find and mangle hwaddress and assign the 'setting' value directly as before, instead of calling functions.

The purpose of the connection_binding_setting function is: give me connection_binding_setting for this bindto parameter. It seems still valid to me.

Reply:
That was only suggestion so I'm fine with both solutions. You can leave it this way if you want.